tallycash/extension
# 🚦 Wallet Router: Implement a wallet router and define a friendly api to wallet providers

# What? 

Implement a wallet provider registry and some rules around switching between them, while being transparent through the `window.ethereum`

# Why?

Because wallet providers are overwriting each other based on confusing environmental conditions, and this is not how this should be handled.

# How? 

See the comments on the commits

# Test

- Install Tally from the dev build and metamask from the chrome store
- go to https://app.uniswap.org/ 
- click connect wallet and see `Injected`
- turn off the default wallet in tally > connect wallet > see `MetaMask`
- turn on the default wallet in tally > connect wallet > see `Injected`
